### PR TITLE
taipyignore file (#501)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
example of .taipyignore file
```
**/*.py*
.*ignore

```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -122,6 +122,7 @@ dataset
 # Demos folder
 demo*/
 !demos/
+.taipyignore
```
Comment (FabienLelaquais) [inline .gitignore]: Really?

**Review by FabienLelaquais — APPROVED** [review]
😍

File: .gitignore (hunk 1 above)
Comment (FredLL-Avaiga) [reply .gitignore]: no

File: src/taipy/gui/server.py [hunk 2]
```diff
@@ -74,6 +75,17 @@ def handle_message(message) -> None:
             elif "type" in message:
                 gui._manage_message(message["type"], message)
 
+    def __is_ignored(self, file_path: str) -> bool:
+        if not hasattr(self, "_ignore_matches"):
+            ignore_file = (pathlib.Path(__main__.__file__).parent / ".taipyignore") if hasattr(__main__, "__file__") else None
```
Comment (dinhlongviolin1) [inline src/taipy/gui/server.py]: You might want to move `.taipyignore` filename as a constant

**Review by dinhlongviolin1 — APPROVED** [review]

File: src/taipy/gui/server.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/server.py]: a variable at least, you're right

**Pull request merged**